### ISSUE TITLE
SNOW-1507519: Support column_order="name" based inserts

### ISSUE THREAD

**Issue by sfc-gh-pkommini** [snowflakedb/snowpark-python#1844]
## What is the current behavior?
While execute this code:

```python
my_temp_table.write.mode("append").save_as_table(
            [self.database, self.schema, self.table],
            column_order='name',
        )
```

We get this error:
```
NotImplementedError: [Local Testing] Inserting data into table by matching columns is not supported.
```

## What is the desired behavior?
I want the insert to happen.

## How would this improve `snowflake-snowpark-python`?
This would allow us to insert without specifying column list.

## References, Other Background
We need this for an internal project.

**Comment by sfc-gh-sghosh** [comment]
Hello @sfc-gh-pkommini ,

Thanks for raising the issue.
This is working fine without local_testing.  The issue is only with local_testing.
We will work on eliminating it.

Regards,
Sujan

**Comment by mraraujo** [comment]
This would also be meaningful for us, especially related to inserts on tables with autoincrement columns.

**Comment by sfc-gh-pkommini** [comment]
@sfc-gh-sghosh Started a PR for this feature #1854

**Comment by sfc-gh-aling** [comment]
PR merged: https://github.com/snowflakedb/snowpark-python/pull/1872, it will be included in our next release

**Comment by sfc-gh-aling** [comment]
closing the issue as the feature is released in v1.20.0
